pkg/proxy/ipvs/proxier.go

```diff
@@ -577,24 +577,40 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 	}
 	ipvsModules := utilipvs.GetRequiredIPVSModules(kernelVersion)
 
+	var bmods []string
+
+	// Check if this is a full static Kernel to bypass the next checks
+	_, err = ioutil.ReadFile("/proc/modules")
+	if err != nil {
```

Comment:
if `err == os.ErrNotExist`?

Reply:
> @rikatz sorry for the delay. I haven't really seen scenarios where `/proc/modules` doesn't exist on the kernel so was doing my own research. Are we sure that `/proc/modules` doesn't exist rather than being an empty file?

Yes, pretty sure. I've done some tests with a Slackware compiled without supporting loadable modules and the /proc/modules doesn't exists.

I can simulate again here and paste the kernel config, the results just to be sure I'm not saying something wrong :)

Reply:
```
 ls -l /proc/modules
/bin/ls: cannot access '/proc/modules': No such file or directory
```
This is a Kernel configured without "Enable loadable module support"

Reply:
Changed